### ISSUE TITLE
"Random" client crash while entering large form

### ISSUE THREAD

**Issue by IanMayo** [serge-web/serge#876]
A number of users have to complete a form that has around 6 textarea fields.

At a random point in time (at a moment when they may not actually be typing) the Serge client crashes, leaving them with a plain blue screen.

Looking at the console in developer tools, it's clear the issue is in the `json-editor` component.

```
Uncaught TypeError. "Cannot ready type 'Property' of null" - jsoneditor.js:5361
Uncaught TypeError. "Cannot ready type 'remove_empty_properties_' of null" - jsoneditor.js:3558
```

I've looked at the JS code at that point, and it's when the JS is referring to the schema.   It looks like the schema is null.

The above is a challenging issue.

But, we may be able to address the "Random" aspect.  It may actually be related to the heartbeat.

I think that last week I learned that the whole UI re-renders on the heartbeat.  This is unnecessary.

Could we please isolate the heart-beat so that only the heart icon gets re-rendered?

**Comment by lilitkarapetyan** [comment]
"I think that last week I learned that the whole UI re-renders on the heartbeat. This is unnecessary." - I can do heartbeat with pure css

**Comment by IanMayo** [comment]
Yes please Lilit.  I _think_ it's already in CSS.  It's just that the CSS modification is performed as part of a re-render.

**Comment by sanjayradadiya** [comment]
Instead of updating in redux we need to directly call to api and get the result and based on that re-render the version component only

**Comment by IanMayo** [comment]
Ok, thanks @sanjayradadiya 

So if we update in redux, the whole UI has to re-render?  I thought react/redux was _intelligent_ in what got re-rendered?

**Comment by sanjayradadiya** [comment]
Yeah, it should be but that will only occurred in pure component or React.memo 

![image](https://user-images.githubusercontent.com/16476120/123633172-80764400-d836-11eb-971f-d501106fa5b6.png)

May be i was updating `serverPingTime` in all call so it is get update all time in redux

**Comment by IanMayo** [comment]
I did some investigating yesterday.  

I saw that while most of the call stack was in `jsoneditor`, the only line in the call stack from our code was `editor.destroy()` at line 63. 

Here is what was at line 63:
![image](https://user-images.githubusercontent.com/1108513/124177819-a60a8400-daa8-11eb-9575-5db269c2c10b.png)

I know the implementation has moved on from the above code, but I'm still worried about some multi-threaded issue related to `editor.destroy()`.  Is it an option for to wrap our usage of `editor.destroy()` in a try/catch block, and to log the errors?
